src/backend/metal/src/device.rs

```diff
@@ -971,11 +970,17 @@ impl hal::Device<Backend> for Device {
                 .expect("no associated vertex buffer found");
             // handle wrapping offsets
             let elem_size = element.format.surface_desc().bits as pso::ElemOffset / 8;
-            let (cut_offset, base_offset) = if element.offset + elem_size <= original.stride {
+            let stride = if original.stride != 0 {
+                original.stride
+            } else {
+                warn!("Zero sized vertex stride found!");
```

Comment:
Is there a Vulkan reference for what's supposed to happen in this case? I don't see it explicitly disallowed

Reply:
Molten just considers the stride of `float4` in this case, so I thought it doesn't matter that much.
But thinking about this more (as well as checking the documentation), the 0 stride has a very clear use case: repeat the attribute. What we should be doing in Metal then is switching it to be per-instance and setting the instance rate to the max.
Perhaps, in a follow-up?